lib/inferno/dsl/fhir_client.rb

```diff
@@ -59,6 +59,13 @@ def fhir_clients
         @fhir_clients ||= {}
       end
 
+      # @private
+      def fhir_error_filter(&block)
```

Comment:
I would rename this to: `fhir_client_tcp_exception_handler`  (and `http_client_tcp_exception_handler`).  Yeah its verbose, but might as well be clear about what it is doing.

Also, another option here would be to have a more generic `tcp_exception_handler` method, put that in another file (similar to request_storage... and require_relative/klass.include it in the same way in both clients), and then have that code have a 2 part rescue.  First, for `SocketErrors`, then a separate rescue for `Faraday:ConnectionFailed`.  That way you can make it so that method is a little less verbose.  One downside to this is that in practice that separate file will have to know how all clients work underneath the hood.  So I'm not that opinionated about it.